lib/rubocop/cop/mixin/hash_slice_except.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for Style/HashExcept and Style/HashSlice
+    module HashSliceExcept
+      include RangeHelp
+      extend NodePattern::Macros
+
+      RESTRICT_ON_SEND = %i[reject select filter].freeze
+
+      # @!method bad_method_with_poro?(node)
+      def_node_matcher :bad_method_with_poro?, <<~PATTERN
+        (block
+          (call _ _)
+          (args
+            $(arg _)
+            (arg _))
+          {
+            $(send
+              _ {:== :!= :eql? :include?} _)
+            (send
+              $(send
+                _ {:== :!= :eql? :include?} _) :!)
+            })
+      PATTERN
+
+      # @!method bad_method_with_active_support?(node)
+      def_node_matcher :bad_method_with_active_support?, <<~PATTERN
+        (block
+          (send _ _)
+          (args
+            $(arg _)
+            (arg _))
+          {
+            $(send
+              _ {:== :!= :eql? :in? :include? :exclude?} _)
+            (send
+              $(send
+                _ {:== :!= :eql? :in? :include? :exclude?} _) :!)
+            })
+      PATTERN
+
+      def on_send(node)
+        raise NotImplementedError
+      end
+
+      private
+
+      def extract_offense(node)
+        block = node.parent
+        return unless bad_method?(block)
+
+        except_key = except_key(block)
+        return if except_key.nil? || !safe_to_register_offense?(block, except_key)
+
+        [offense_range(node), except_key_source(except_key)]
+      end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+      def bad_method?(block)
```

Comment:
I know this is copied from `HashExcept` but I'd like to try renaming `bad_method?` to something more descriptive. By extension, this goes for the two matchers as well (`bad_method_with_active_support?` and `bad_method_with_poro?`). 

Perhaps `extracts_hash_subset?`? @Earlopain any ideas?

Reply:
I noticed that as well but was going to tackle it in a separate refactor after this had gotten merged to keep from scope creep. If you want to tackle the refactor, I think it'd be best to actually combine the two pattern matchers which are almost identical except for which methods are allowed, and handle that within what is currently `bad_method?` instead. I think we need tests for the Rails methods when ActiveSupportExtensions is not enabled to be added first too.

Reply:
Here's my attempt at this: https://github.com/rubocop/rubocop/pull/13517